### ISSUE TITLE
[BUG]: errors loading Dev web app in the browser 

### ISSUE THREAD

**Issue by coglmn** [activepieces/activepieces#3510]
**Describe the bug**

Installing Activepieces on a Linux VM. The code compiles in the Dev mode successfully with 'npm start', however, the Web app does not load in the browser, and raises the following error:

ERROR Error: Uncaught (in promise): HttpErrorResponse: {"headers":{"normalizedNames":{},"lazyUpdate":null,"headers":{}},"status":0,"statusText":"Unknown Error","url":"http://localhost:3000/v1/flags","ok":false,"name":"HttpErrorResponse","message":"Http failure response for http://localhost:3000/v1/flags: 0 Unknown Error","error":{"isTrusted":true}}
    at resolvePromise (zone.js:1193:31)
    at resolvePromise (zone.js:1147:17)
    at zone.js:1260:17
    at _ZoneDelegate.invokeTask (zone.js:402:31)
    at core.mjs:10735:55
    at AsyncStackTaggingZoneSpec.onInvokeTask (core.mjs:10735:36)
    at _ZoneDelegate.invokeTask (zone.js:401:60)
    at Object.onInvokeTask (core.mjs:11048:33)
    at _ZoneDelegate.invokeTask (zone.js:401:60)
    at Zone.runTask (zone.js:173:47)
handleError @ core.mjs:10592
Show 1 more frame
Show less
localhost:3000/v1/flags:1 
    
       Failed to load resource: net::ERR_CONNECTION_REFUSED

**To Reproduce**

1. Clone the repo from github on a linux machine
2. update the environment variables in packages/backend/.env to change 'localhost' to 0.0.0.0:  (given the Unix VM is not local machine, it would be running in the cloud)

AP_WEBHOOK_URL="http://0.0.0.0:3000"
AP_FRONTEND_URL="http://0.0.0.0:4200"

3. run 'npm start'. It successfully compiles and at the end shows that the server is listening at the following addresses:
 
Server listening at http://0.0.0.0:3000
The application started on http://0.0.0.0:4200, as specified by the AP_FRONTEND_URL variables.

4. In the browser, now typing https://xx.xx.xx.xx:4200 in which xx.xx.xx.xx is the public IP of the machine loads up the app (meaning the mapping from IP to 0.0.0.0 works) and port 4200 is open. It attempts to load the page (shows the spinning wheel), however, raises the following error in the console:

ERROR Error: Uncaught (in promise): HttpErrorResponse: {"headers":{"normalizedNames":{},"lazyUpdate":null,"headers":{}},"status":0,"statusText":"Unknown Error","url":"http://localhost:3000/v1/flags","ok":false,"name":"HttpErrorResponse","message":"Http failure response for http://localhost:3000/v1/flags: 0 Unknown Error","error":{"isTrusted":true}}
    at resolvePromise (zone.js:1193:31)
    at resolvePromise (zone.js:1147:17)
    at zone.js:1260:17
    at _ZoneDelegate.invokeTask (zone.js:402:31)
    at core.mjs:10735:55
    at AsyncStackTaggingZoneSpec.onInvokeTask (core.mjs:10735:36)
    at _ZoneDelegate.invokeTask (zone.js:401:60)
    at Object.onInvokeTask (core.mjs:11048:33)
    at _ZoneDelegate.invokeTask (zone.js:401:60)
    at Zone.runTask (zone.js:173:47)
handleError @ core.mjs:10592
Show 1 more frame
Show less
localhost:3000/v1/flags:1 
        
        
       Failed to load resource: net::ERR_CONNECTION_REFUSED

It seems that the server is trying to connect to localhost:3000 instead of 0.0.0.0:3000 which the server is running on it (and specified in the Env variable). 

And, if I type http://xx.xx.xx.xx:3000 the backend server in the log shows that it is called.... so that address/mapping also works.

Looking at the code, there are many places where localhost:3000 is hard-coded, where I suspect the address that the UI is using to call the backend is coming from.  When accessed from the browser, localhost would point to the client machine where the browser is running on, not the remote VM.

**Expected behavior**
Given, we have environment variables, the expectation is that it reads the URL addresses from the Env rather than from the code.

Is this the right way to run the dev UI / Backend in Dev mode on a remote VM, or there are other / better ways?

**Screenshots**
![image](https://github.com/activepieces/activepieces/assets/44345688/014b0622-1d96-44cd-b1f2-7e4982b69a0c)

**Comment by abuaboud** [comment]
I would recommend use Dev Containers (Check the documentation) in that case it's more consistent.

Another way try to change `environment.ts` in packages/ui/common/src/lib/environemnt/environment.ts change the apiUrl there from localhost, and let me know if that works.

**Comment by A-VIKASH-KUMAR** [comment]
The server running on http endpoint but u trying to hit with https endpoint which causing connection refused error, please try this if this works

**Comment by coglmn** [comment]
> packages/ui/common/src/lib/environemnt/environment.ts

Hi @abuaboud  Thank you. That was a good hint. I updated the apiURL in environment.ts initially replacing 'localhost' with '0.0.0.0'. however, it did not work. Then, I replaced it with the public IP of the VM where the code is running on it, and it loaded the login page correctly.  For dev/test, this is OK. For more systematic work, Dev container is the way go perhaps. Will try that next.
